### ISSUE TITLE
[Bug]: did-create-window does not fire for window.open calls with a re-used target

### ISSUE THREAD

**Issue by thorsent** [electron/electron#36949]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

22.0.0

### What operating system are you using?

Windows

### Operating System Version

Confirmed on Windows 10 and MacOS Monterey

### What arch are you using?

x64

### Expected Behavior

When `window.open("http://google.com", "blah")` is called from a renderer window, we expect a `setWindowOpenHandler` handler to be called (system level), and then for a `did-create-window` event to trigger (on the webContents of the parent window).

### Actual Behavior

This sequence occurs as expected for the first call, but then subsequent calls fail to trigger the `did-create-window` event (though they do call setWindowOpenHandler).

An alternative solution would be for the webContentId to be recycled when the window.open target is re-used. This would allow the parent window to continue to manage the child window.

### Testcase Gist URL

[gist](https://gist.github.com/thorsent/605177dd2a7b16b870486c1191cb0131)

### Additional Information

This behavior is important because the child window's webContentsId changes with each window.open call–even when using the same target. Without a `did-create-window` event, there is no mechanism for the parent window to receive the new webContentsId (and thus no way to manage that window).

**Comment by thorsent** [comment]
Electron fiddle gist reproducer: https://gist.github.com/thorsent/605177dd2a7b16b870486c1191cb0131

**Comment by pushkin-** [comment]
Seems similar to this issue https://github.com/electron/electron/issues/29880

**Comment by codebytere** [comment]
@thorsent this can be addressed by controlling for `frameName` - in the renderer process of your fiddle:

```
let count = 0
addEventListener('DOMContentLoaded', (event) => {
    document.getElementsByTagName("button")[0].addEventListener("click", () => {
        window.open("http://google.com", `blah${count++}`)
    })
})
```

does that solve for your use case?

**Comment by thorsent** [comment]
Our UX calls for the window to be re-used, so changing the frame name doesn't work because it creates new windows.

We did find a workaround. We store the names of all windows that are opened. Then, in setWindowOpenHandler, if the name already exists (found in details) then we prevent a new window from being created and instead call loadUrl.

I think this probably breaks the window reference in the browser (`var w = window.open()`) so it is not ideal compared with a real event that returns the new webContents, but it does unblock us.

This ticket can be closed though I would recommend at least adding a note to the docs for setWindowEventHandler and/or did-create-window, and to consider whether an event ought to be triggered on webContents for this condition (perhaps webcontents-created could be triggered on the parent webContents as a more generalized way of tracking a chain).

**Comment by nornagon** [comment]
Unfortunately, Chrome treats `window.open` with an existing frame as a _navigation_, not a new window, so we don't get the normal events that would trigger `setWindowOpenHandler`.

Theoretically, you should be able to use the `will-navigate` or `did-navigate` events on the child window's WebContents to detect `window.open` from the parent window, but at present it's impossible in Electron to distinguish between a navigation initiated by the child window and a navigation initiated by the parent window. This should be fixed! I have a draft PR (#37085), pending API improvements.

**Comment by nornagon** [comment]
I'm going to close this because I feel #37085 is a feature, and this system is otherwise working as intended.
